src/components/shared/filterDropdownButton/dateFieldGroup.js

```diff
@@ -0,0 +1,163 @@
+import { withTranslation } from 'react-i18next';
+import React from 'react';
+import moment from 'moment';
+import { firstBlockTime } from '../../../constants/datetime';
+import { getDateTimestampFromFirstBlock, formatInputToDate } from '../../../utils/datetime';
+import { getInputSelection, setInputSelection } from '../../../utils/selection';
+import DateField from './dateField';
+import Feedback from '../../toolbox/feedback/feedback';
+import styles from './filters.css';
+
+class DateFieldGroup extends React.Component {
+  constructor(props) {
+    super();
+
+    this.state = {
+      fields: {
+        dateTo: {
+          error: false,
+          value: '',
+          loading: false,
+        },
+        dateFrom: {
+          error: false,
+          value: '',
+          loading: false,
+        },
+      },
+      feedback: '',
+    };
+
+    this.inputRefs = {
+      dateTo: null,
+      dateFrom: null,
+    };
+
+    this.timeout = null;
+
+    this.dateFormat = props.t('DD.MM.YY');
+
+    this.handleFieldChange = this.handleFieldChange.bind(this);
+    this.setInputRefs = this.setInputRefs.bind(this);
+  }
+
+  /* istanbul ignore next */
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
+  }
+
+  setInputRefs(node) {
+    if (node && node.name) {
+      this.inputRefs[node.name] = node;
+    }
+  }
+
+  validateDates(fieldsObj, selectionObj) {
+    const { t } = this.props;
+    let feedback = '';
+    // eslint-disable-next-line max-statements
+    const fields = Object.keys(fieldsObj).reduce((acc, field) => {
+      const value = fieldsObj[field].value || '';
+      const date = moment(value, this.dateFormat);
+      let error = false;
+
+      if (value && !date.isValid()) {
+        feedback = t(`Date must be in ${this.dateFormat} format`);
+        error = true;
+      } else if (
+        (field === 'dateFrom' && date > moment(fieldsObj.dateTo.value, this.dateFormat))
+        || (field === 'dateTo' && date < moment(fieldsObj.dateFrom.value, this.dateFormat))
+      ) {
+        feedback = t('Invalid dates');
+        error = true;
+      }
+
+      if (date.isValid() && getDateTimestampFromFirstBlock(value, this.dateFormat) < 0) {
+        feedback = t('Date must be after {{firstBlock}}', {
+          firstBlock: moment(firstBlockTime).format(this.dateFormat),
+        });
+        error = true;
+      }
+
+      return {
+        ...acc,
+        [field]: {
+          value,
+          error,
+          loading: false,
+        },
+      };
+    }, {});
```

Comment:
Could be possible to refactor this part of the function to more the eslint ignore flag?
and maybe you don't need to set the error in every if you can do something like this.

Instead of 

```
feedback = t(`Date must be in ${this.dateFormat} format`);
error = true;
```

you can set the error based on the feedback

```
return {
        ...acc,
        [field]: {
          value,
          error: !!feedback,
          loading: false,
        },
      };
    }, {});
```